runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -907,19 +908,19 @@ TR_J9SharedCache::rememberClass(J9Class *clazz, const AOTCacheClassChainRecord *
    if (chainLength > maxClassChainLength * sizeof(uintptr_t))
       {
       LOG(1, "\t\t > %u so bailing\n", maxClassChainLength);
-      return NULL;
+      return TR_SharedCache::INVALID_CLASS_CHAIN_OFFSET;
       }
 
    if (!fillInClassChain(clazz, chainData, chainLength, numSuperclasses, numInterfaces))
       {
       LOG(1, "\tfillInClassChain failed, bailing\n");
-      return NULL;
+      return TR_SharedCache::INVALID_CLASS_CHAIN_OFFSET;
       }
 
    if (!create)
       {
       LOG(1, "\tnot asked to create but could create, returning non-null\n");
-      return (uintptr_t *)0x1;
+      return 1;
```

Comment:
I think that the only time `create` is false in a `rememberClass` call is when `canRememberClass` or one of its overloads is run:

https://github.com/eclipse-openj9/openj9/blob/00a123d316ab18a634e941a429c5a2962d912cdd/runtime/compiler/env/J9SharedCache.hpp#L176-L179

In those cases we don't care about the offset at all, and the `return 1` is just a method of communicating that the chain could have been created. It looks like the intent is to avoid duplicating the `rememberClass` code in `canRememberClass`.

I'm not sure if 1 could be a valid offset - judging from `isPointerInSharedCache`:

https://github.com/eclipse-openj9/openj9/blob/00a123d316ab18a634e941a429c5a2962d912cdd/runtime/compiler/env/J9SharedCache.cpp#L734

that would happen only if a class chain started at exactly the `metadataStartAddress` of the first layer of an SCC.

Reply:
I'll add a symbolic constant.